src/main/flight/mixer.c

```diff
@@ -218,22 +218,38 @@ static void calculateThrottleAndCurrentMotorEndpoints(timeUs_t currentTimeUs)
         }
     } else {
         throttle = rcCommand[THROTTLE] - PWM_RANGE_MIN + throttleAngleCorrection;
+
 #ifdef USE_DYN_IDLE
-        if (mixerRuntime.idleMinMotorRps > 0.0f) {
-            const float maxIncrease = isAirmodeActivated() ? mixerRuntime.idleMaxIncrease : 0.04f;
-            const float minRps = rpmMinMotorFrequency();
-            const float targetRpsChangeRate = (mixerRuntime.idleMinMotorRps - minRps) * currentPidProfile->idle_adjustment_speed;
-            const float error = targetRpsChangeRate - (minRps - mixerRuntime.oldMinRps) * pidGetPidFrequency();
-            const float pidSum = constrainf(mixerRuntime.idleP * error, -currentPidProfile->idle_pid_limit, currentPidProfile->idle_pid_limit);
-            motorRangeMinIncrease = constrainf(motorRangeMinIncrease + pidSum * pidGetDT(), 0.0f, maxIncrease);
-            mixerRuntime.oldMinRps = minRps;
-            throttle += mixerRuntime.idleThrottleOffset * RC_COMMAND_THROTTLE_RANGE;
-
-            DEBUG_SET(DEBUG_DYN_IDLE, 0, motorRangeMinIncrease * 1000);
-            DEBUG_SET(DEBUG_DYN_IDLE, 1, targetRpsChangeRate);
-            DEBUG_SET(DEBUG_DYN_IDLE, 2, error);
-            DEBUG_SET(DEBUG_DYN_IDLE, 3, minRps);
-        } else {
+        if (mixerRuntime.dynIdleMinRps > 0.0f) {
+            const float maxIncrease = isAirmodeActivated() ? mixerRuntime.dynIdleMaxIncrease : 0.04f;
+            float minRps = rpmMinMotorFrequency();
+            DEBUG_SET(DEBUG_DYN_IDLE, 3, (minRps * 10));
+            const float rpsError = mixerRuntime.dynIdleMinRps - minRps;
+
+            float dynIdleP = rpsError * mixerRuntime.dynIdlePGain;
+
+            minRps = mixerRuntime.prevMinRps + 0.01f * (minRps - mixerRuntime.prevMinRps);
+            // PT1 lowpass smoothing provides smoothing and delay for D
+            // 0.01 is 6hz on 4k, 25ms delay, or 67hz at 8k with 2.5ms delay
+            // *** TODO *** correct for DTc, use proper pt1
+            float dynIdleD = (mixerRuntime.prevMinRps - minRps) * mixerRuntime.dynIdleDGain;
+
+            if (mixerRuntime.motorsIncreased) {
+                mixerRuntime.dynIdleI += (dynIdleD * 60.0f + rpsError * mixerRuntime.dynIdleIGain) * pidGetDT();
+            }
+            mixerRuntime.dynIdleI = constrainf(mixerRuntime.dynIdleI, 0.0f, maxIncrease);
+            // stop I being dragged hard negative when rps is high, hold recent value as usually closer than any other value
+
+            motorRangeMinIncrease = constrainf((dynIdleP + mixerRuntime.dynIdleI + dynIdleD), 0.0f, maxIncrease);
+            mixerRuntime.motorsIncreased = (motorRangeMinIncrease > 0);
+
+//            DEBUG_SET(DEBUG_DYN_IDLE, 0, (motorRangeMinIncrease * 10000));
```

Comment:
Please remove commented out code.

Reply:
OK will do

Reply:
I'll leave in the comments that require action before the code is complete, or are necessary for a reviewer to appreciate why something has been done.